.github/workflows/binaries.yml

```
@@ -11,11 +11,9 @@ jobs:
       - uses: actions/checkout@v4
       - id: generate
         run: |
-          cat >>"$GITHUB_OUTPUT" <<EOM
-          matrix<<EOJSON
-          $(./.github/bin/build-binaries-matrix)
-          EOJSON
-          EOM
+          set -Eeu
+          ./.github/bin/build-binaries-matrix > matrix
+          echo "matrix=$(cat matrix)" >> "$GITHUB_OUTPUT"
```

Sorry I reviewed mid-draft. I'll leave this as a formal suggestion now:

```suggestion
          ./.github/bin/build-binaries-matrix > matrix
          cat >>"$GITHUB_OUTPUT" <<EOM
          matrix<<EOJSON
          $(cat matrix)
          EOJSON
          EOM
```